File: q30-q39/q34/q34.cpp
class Solution {
public:
    vector<int> searchRange(vector<int>& nums, int target) {
        int left = 0, right = nums.size() - 1;
        int middle = 0;
        bool found = false;
        while (left <= right){
            middle = (left + right) / 2;
            if (nums[middle] == target){
                found = true;
                break;
            }
                
            if (nums[middle] <= target){
                left = middle + 1;
            } else {
                right = middle -1;
            }  
        }
        if (found){
            left = middle;
            right = middle;
            while (left - 1 >= 0 && nums[left-1] == target){
                --left;
            }
            while (right + 1 <= nums.size()-1 && nums[right+1] == target){
                ++right;
            }
            std::vector<int> solution;
            solution.push_back(left);
            solution.push_back(right);
            return solution;
        } 
        std::vector<int> solution;
        solution.push_back(-1);
        solution.push_back(-1);
        return solution;
    }
};

File: Sequence/q1-q99/q30-q39/q34/q34.py
class Solution:
    def searchRange(self, nums: List[int], target: int) -> List[int]:
        left = 0
        right = len(nums) - 1
        found = False
        while left <= right:
            middle = (left + right) // 2
            if nums[middle] == target:
                found = True
                break
            if nums[middle] <= target:
                left = middle + 1
            else:
                right = middle - 1
        if (found):
            left = middle
            right = middle
            while left - 1 >= 0 and nums[left - 1] == target:
                left = left -1
            while right + 1 <= len(nums) -1 and nums[right + 1] == target:
                right = right + 1
            return [left, right]
        return [-1, -1]
